src/common/row_operations/row_gather.cpp

```diff
@@ -111,4 +116,74 @@ void RowOperations::Gather(Vector &rows, const SelectionVector &row_sel, Vector
 	}
 }
 
+template <class T>
+static void TemplatedFullScanLoop(Vector &rows, Vector &col, idx_t count, idx_t col_offset, idx_t col_no) {
+	// Precompute mask indexes
+	idx_t entry_idx;
+	idx_t idx_in_entry;
+	ValidityBytes::GetEntryIndex(col_no, entry_idx, idx_in_entry);
+
+	auto ptrs = FlatVector::GetData<data_ptr_t>(rows);
+	auto data = FlatVector::GetData<T>(col);
+	auto &col_mask = FlatVector::Validity(col);
+
+	for (idx_t i = 0; i < count; i++) {
+		auto row = ptrs[i];
+		data[i] = Load<T>(row + col_offset);
+		ValidityBytes row_mask(row);
+		if (!row_mask.RowIsValid(row_mask.GetValidityEntry(entry_idx), idx_in_entry)) {
+			col_mask.SetInvalid(i);
```

Comment:
This line is uncovered. Could you add a test that covers this?

Reply:
For this line to be hit we need null values in the hash table.
However null comparisons are not done in this join, they are skipped in a phase prior to that.
I don't think we should remove this code, because at some point we might implement it anyway, but it is "unreachable" through testing right now. What you think?

Reply:
Perhaps turn this into an internal exception for now if it is never supposed to be triggered currently, with the line of code commented out? If we need it in the future we can remove the exception again.